File: 0682 baseball-game/solution.java
class Solution {
    public int calPoints(String[] o) {
        int[] st = new int[o.length];
        int sum=0,top=-1;
        for(String s:o) {
            if(s.equals("C")) top--;
            else if(s.equals("D")) {
                int n=st[top];
                st[++top]=n*2;
            } else if(s.equals("+")) {
                st[++top]=st[top-2]+st[top-1];
            } else 
               st[++top]=Integer.parseInt(s);
        }
        for(int i=0;i<=top;i++) sum+=st[i];
        return sum;
    }
}

File: 0682 baseball-game/solution.ts
function calPoints(operations: string[]): number {
    let st = [];
    let top=-1;
    for(const a of operations) {
        if(top!=-1 && a=="+") {
            st[++top] = st[top-1] + st[top-2];
        } else if(top!=-1 && a=='D') {
            let r = st[top];
            st[++top] = r*2;
        } else if(top!=-1 && a=='C') {
            top--;
        } else {
            st[++top] = Number(a);
        }
    }
    let sum = 0;
    for(let i=0;i<=top;i++) sum+=st[i];
    return sum;
};
